records_mover/records/validated_records_hints.py

```diff
@@ -10,142 +12,175 @@
                     VALID_DATETIMEFORMATTZS, VALID_DATETIMEFORMATS)
 from .hints import cant_handle_hint
 
-
-# TODO: Read up on namedtuple vs dataclass
-# TODO: Read up on autovalidating libraries
-class ValidatedRecordsHints(NamedTuple):
-    header_row: HintHeaderRow
-    field_delimiter: HintFieldDelimiter
-    compression: HintCompression
-    record_terminator: HintRecordTerminator
-    quoting: HintQuoting
-    quotechar: HintQuoteChar
-    doublequote: HintDoublequote
-    escape: HintEscape
-    encoding: HintEncoding
-    dateformat: HintDateFormat
-    timeonlyformat: HintTimeOnlyFormat
-    datetimeformattz: HintDateTimeFormatTz
-    datetimeformat: HintDateTimeFormat
-
-    @staticmethod
-    def validate(hints: RecordsHints,
-                 fail_if_cant_handle_hint: bool) -> 'ValidatedRecordsHints':
+HintT = TypeVar('HintT')
+
+
+HintName = Literal["header-row",
+                   "field-delimiter",
+                   "compression",
+                   "record-terminator",
+                   "quoting",
+                   "quotechar",
+                   "doublequote",
+                   "escape",
+                   "encoding",
+                   "dateformat",
+                   "timeonlyformat",
+                   "datetimeformattz",
+                   "datetimeformat"]
+
+
+class Hint(Generic[HintT]):
+    def __init__(self,
+                 type_: Type[HintT],
+                 hint_name: HintName,
+                 default: HintT):
+        self.default = default
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker